soroban-simulation/src/lib.rs

```diff
@@ -0,0 +1,315 @@
+mod fees;
+mod ledger_storage;
+mod state_ttl;
+
+use anyhow::{anyhow, bail, Context, Result};
+use ledger_storage::LedgerStorage;
+use soroban_env_host::auth::RecordedAuthPayload;
+use soroban_env_host::budget::Budget;
+use soroban_env_host::events::Events;
+use soroban_env_host::storage::Storage;
+use soroban_env_host::xdr::{
+    AccountId, ConfigSettingEntry, ConfigSettingId, DiagnosticEvent, InvokeHostFunctionOp,
+    LedgerFootprint, LedgerKey, OperationBody, ScVal, SorobanAddressCredentials,
+    SorobanAuthorizationEntry, SorobanCredentials, SorobanTransactionData, VecM,
+};
+use soroban_env_host::{DiagnosticLevel, Host, LedgerInfo};
+use std::collections::HashSet;
+use std::convert::{TryFrom, TryInto};
+use std::iter::FromIterator;
+use std::rc::Rc;
+
+pub struct RestorePreamble {
+    pub transaction_data: SorobanTransactionData,
+    pub min_fee: i64,
+}
+
+#[derive(Default)]
+pub struct PreflightResult {
```

Comment:
With the crate rename it would probably make sense to make a find-and-replace pass to change 'Preflight'  into `Simulation/Simulate` depending on context.

Reply:
Done